src/main/java/com/okta/tools/authentication/OktaMFA.java

```diff
@@ -120,13 +126,22 @@ private static JSONObject selectFactor(JSONObject primaryAuthResponse) throws JS
         }
 
         if (supportedFactors.size() > 1) {
+            if (environment.oktaMfaChoice != null) {
+                for (JSONObject factor : supportedFactors) {
+                    String provider = factor.getString("provider");
+                    String factorType = factor.getString("factorType");
+                    if ((provider + "." + factorType).equals(environment.oktaMfaChoice)) {
+                        return factor;
+                    }
+                }
+            }
             System.err.println("\nMulti-Factor authentication is required. Please select a factor to use.");
             System.err.println("Factors:");
             for (int i = 0; i < supportedFactors.size(); i++) {
                 JSONObject factor = supportedFactors.get(i);
-                factorType = factor.getString("factorType");
-                factorType = getFactorDescription(factorType, factor);
-                System.err.println("[ " + (i + 1) + " ] : " + factorType);
+                String factorType = factor.getString("factorType");
+                String factorDescription = getFactorDescription(factorType, factor);
+                System.err.println("[ " + (i + 1) + " ] : " + factorDescription);
```

Comment:
:art: moved scope of variables inward and gave distinct states different variable names.